web/modules/pocketbase/types/types.ts

```diff
@@ -0,0 +1,93 @@
+import type { ListResult, RecordAuthResponse, RecordModel } from 'pocketbase'
+
+/**
+ * Base client interface for database operations
+ * Provides methods for CRUD operations on items and user authentication
+ */
+export interface BaseClient {
+  /**
+   * Retrieves a single item by its unique identifier
+   * @param id - The unique identifier of the item to retrieve
+   * @returns Promise resolving to the item record
+   */
+  getItem: (id: string) => Promise<RecordModel>
```

Comment:
I think these methods should no return RecordModel, but rather an internal type, which can be used throughout the application. And as for the type of the items below, these types are for this project only and can't be used for any other project, so why not return them in these methods? And all transformations would be done internally in pocketbase client without exposing them to the consumer.

What I mean is that this method should be:

```suggestion
  getItem: (id: string) => Promise<Item>
```

Or if you wanted to make fully independent pocketbase client, than this module should not contain Frontmatter, DebtTransaction, etc.

Personally I would prefer the one I suggested above.